backstage-plugin/plugins/open-dora/src/components/DashboardComponent/DashboardComponent.tsx

```diff
@@ -9,26 +9,43 @@ import {
 import { useApi } from '@backstage/core-plugin-api';
 import { getEntityRelations, useEntity } from '@backstage/plugin-catalog-react';
 import { Grid } from '@material-ui/core';
-import React, { useEffect } from 'react';
+import React, { useEffect, useReducer } from 'react';
 import { MetricData } from '../../models/MetricData';
 import { groupDataServiceApiRef } from '../../services/GroupDataService';
 import { BarChartComponent } from '../BarChartComponent/BarChartComponent';
 import { DropdownComponent } from '../DropdownComponent/DropdownComponent';
 import './DashboardComponent.css';
+import { ChartErrors } from '../../models/CustomErrors';
 
 export interface DashboardComponentProps {
   entityName?: string;
   entityGroup?: string;
 }
+
+function dataErrorReducer(
+  currentErrors: ChartErrors,
+  action: { type: keyof ChartErrors; error: Error },
+) {
+  return {
+    ...currentErrors,
+    [action.type]: action.error,
+  };
+}
+
 export const DashboardComponent = ({
   entityName,
   entityGroup,
 }: DashboardComponentProps) => {
   const [chartData, setChartData] = React.useState<MetricData | null>(null);
+  const [chartDataAverage, setChartDataAverage] =
+    React.useState<MetricData | null>(null);
   const [selectedTimeUnit, setSelectedTimeUnit] = React.useState('weekly');
-  const [dataError, setDataError] = React.useState<Error | undefined>(
-    undefined,
-  );
+
+  const initialErrors: ChartErrors = { countError: null, averageError: null };
+
+  // const [dataError, setDataError] = React.useState<ChartErrors>(initialErrors);
```

Comment:
We don't need this anymore

Reply:
the comented out bit? ya..
as for initial errors they are used for reducer on line 48

Reply:
Indeed just the comment. We could also inline initial errors since we don't need it to be a separate const, but we can leave that for now.